proposals/NNNN-HID-Support-Plug-in.md

```diff
@@ -0,0 +1,43 @@
+
+##Proposed solution
+
+Premise condition: "High level UI/Widget library" which support Android/iOS widget for rendering video streaming application in SDL proxy. ex. Android:Virtual Display Encoder, iPhone:XXXX This library can retrieve High level UI/Widget layout information and create High level UI/Widget event.
+This proposal architecture: Adding "Remote Human Interface Device manager" to SDL proxy (Android/iPhone). This module will provide following functionalities.
+dynamic selection of proper plugin for connected HU. This functionalitiy will be implemented with using VehicleType and reflection of Android Java and iOS ObjectiveC/Swift.
+bridge from/to "High-level UI/Widget library" to/from "plug-in module" which is proxy of HU special device.
+Adding "Plug-in for each OEM's specific device" which will be provided by each OEMs. This module may realize following functionalities.
+converting low level event which is sent from HU driver as OnTouchEvent(existing RPC) and/or OnSytemRequest(existing RPC, if OnTouchEvent's information is not enough to realize special event), to High level UI/Widget event, if required.
+sharing High level UI/Widget layout information with HU driver with using OnSystemRequest and/or other RPCs (existing RPC). This information may be used for haptic feedback and/or focus management in HU.
+superimposing information on application rendering image. (ex. cursor, focus highlight). These features requirements depend on each specific device features and OEM's UI/UX policy. NOTE) These feature may not be implemented in plugin. NOTE) Some features may be implemented in HU. NOTE) In some case of OEM specific plug-in, existing RPC may be used for not original purpose. However, this is closed spec between specific device and specific plug-in. So, it's not be affect to SDL general spec.
+
+##Potential downsides
+
+Downside:
+OEM should prepare specific plug-in module for their specific device. If not prepared then, downside imapct is only "the special device will not be applicable use. Full functionality of the device cannot be realized". However, this proposal includes by-pass plug-in which pass thru event. So, if all of event can be mapped to existing RPC (OnTouchEvent) then, applications can be run as minimum without specific plug-in. Of course, no special device and only touch-panel I/F then application behavior is not different.
+The applications not to use High-level UI/Widget library but to use application's UI library, cannot take benefit of this architecture. However, OEM can disclose each plug-in spec and such application may be able to customize to use plug-in feature directly.
+If some specific device behavior cannot be mapped existing RPCs then, still new RPC may be required.
```

Comment:
Please change to "...cannot be mapped to existing RPCs, then a new RPC may still be required."